src/main/service/subscriptionService.ts

```diff
@@ -1,16 +1,42 @@
 import moment from 'moment';
 import {SubscriptionRequests} from '../resources/requests/subscriptionRequests';
 import {CaseSubscription} from '../models/caseSubscription';
+import {Subscription} from '../models/subscription';
 
 const subscriptionRequests = new SubscriptionRequests();
 
 export class SubscriptionService {
 
-  generateCaseTableRows(userid: number): any[] {
-    const subscriptionData = subscriptionRequests.getUserSubscriptions(userid);
+  public async generateSubscriptionsTableRows(userid: number): Promise<any> {
+    const subscriptionData = await subscriptionRequests.getUserSubscriptions(userid);
+    const cases = {
+      cases: [],
+      courts: [],
+    };
+    if (subscriptionData) {
+      cases.cases = this.generateCaseTableRows(subscriptionData);
+      cases.courts = this.generateCourtTableRows(subscriptionData);
+    }
+
+    return cases;
+  }
+
+  public async getSubscriptionUrnDetails(urn: string): Promise<CaseSubscription> {
+    const subscriptions = await subscriptionRequests.getSubscriptionByUrn(urn);
+
+    if (subscriptions) {
+      return subscriptions;
+    } else {
+      console.log(`Subscription with urn ${urn} does not exist`);
+      return null;
+    }
+  }
```

Comment:
This method has come from nowhere and provides no use, we get subscriptions from the method above.
additionally this method is not doing what you seem to think its doing, as you have placed the call in the wrong place, as mentioned in previous comment

Reply:
this code was already there from previous tickets.

Reply:
this method is used by the search urn and it is doing what t is supposed to do. If we want to refactor I have not objections but I think it should be part of an other PR. This is to retrieve the subscriptions on subscription-management page and this method is used on the subscription-urn-search